File: Kth largest element in BST[09-September-2023]/Solution.cpp
//  >> CPP CODE <<

class Solution {
public:
    void solve(Node* root, int& ans, int& k) {
        if (root == nullptr) return;

        solve(root->right, ans, k);

        k--;
        if (k == 0) ans = root->data;

        solve(root->left, ans, k);
    }

    int kthLargest(Node* root, int K) {
        int ans = -1;
        solve(root, ans, K);
        return ans;
    }
};

File: Kth largest element in BST[09-September-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    def kthLargest(self,root, k):
        def solve(root,ans,k):
            if(not root): return
        
            solve(root.right,ans,k)
            
            k[0]-=1
            
            if(k[0]==0): ans[0]=root.data
            
            solve(root.left,ans,k)
        
        ans=[-1]
        solve(root,ans,[k])
        
        return ans[0]
